src/OneExplorer/OneExplorer.ts

```diff
@@ -139,19 +139,31 @@ enum NodeType {
 
 abstract class Node {
   abstract readonly type: NodeType;
-  childNodes: Node[];
+  protected _childNodes: Node[]|undefined;
   uri: vscode.Uri;
 
   abstract icon: vscode.ThemeIcon;
   abstract openViewType: string|undefined;
   abstract canHide: boolean;
 
   constructor(uri: vscode.Uri) {
-    this.childNodes = [];
+    this._childNodes = undefined;
     this.uri = uri;
   }
 
-  abstract buildChildren: () => void;
+  /**
+   * Build `_childNodes` on demand, which is initially undefined
+   */
+  abstract _buildChildren: () => void;
+
+  getChildren(): Node[] {
+    if (this._childNodes) {
+      return this._childNodes;
+    }
+
+    this._buildChildren();
```

Comment:
While I was reviewing this line, this file was merged. :-)

`getChilldren()` seems immutable (since it is getter)
but it does mutable work by calling `_buildChildren()`.

I believe that `_buildChildren()` should not be inside this method.